activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -24,13 +24,19 @@ def visit_Arel_Nodes_DeleteStatement(o, collector)
 
           if has_join_sources?(o)
             collector << "DELETE "
+
             visit o.relation.left, collector
             collector << " FROM "
           else
-            collector << "DELETE FROM "
+            collector << "DELETE "
+            if o.instance_values["relation"].instance_values["table_alias"] && collector&.instance_values["delegate"]
```

Comment:
This makes me think that we might need a tweak and a corresponding test for the case above where the delete has a join